src/utils/ipgen.c

```diff
@@ -50,12 +50,12 @@ void ipgen_free_context(struct ipgenctx *context) {
   }
 }
 
-int run_ip(char *path, char *const argv[]) {
+int run_ip(char *path, const char *const argv[]) {
   return run_argv_command(path, argv, NULL, NULL);
 }
 
 int ipgen_new_interface(char *path, char *ifname, char *type) {
-  char *argv[7] = {"link", "add", "name", ifname, "type", type, NULL};
+  const char *argv[7] = {"link", "add", "name", ifname, "type", type, NULL};
   return run_ip(path, argv);
 }
```

Comment:
I've already fixed all of these in https://github.com/nqminds/edgesec/commit/5599ac6f605651722634b1e08f4a7b69352f0b78, so we might get some merge conflicts here.

It might be easier to use a `git rebase -i main` and just delete https://github.com/nqminds/edgesec/pull/319/commits/0aed6030e7d5d94989fbe73f6803312893f80426, rather than have to worry about fixing all the merge conflicts.

Reply:
I will resolve the conflicts.